src/datahike/tools.cljc

```diff
@@ -35,3 +36,35 @@
     (list `(log/log! :error :p ~fragments ~{:?line (:line (meta &form))})
           `(throw #?(:clj  (ex-info (str ~@(map (fn [m#] (if (string? m#) m# (list 'pr-str m#))) msgs)) ~data)
                      :cljs (error (str ~@(map (fn [m#] (if (string? m#) m# (list 'pr-str m#))) msgs)) ~data))))))
+
+(defn throwable-promise
```

Comment:
Adding this is ok, but it would be good to refer here exactly to where it was coming from (and make sure it is under a compatible licence, i.e. EPL v1 compatible). We might replace it with a `promise-chan` later.